py_mmd_tools/odajson_to_mmd.py

```diff
@@ -0,0 +1,335 @@
+"""
+Tool for creating metadata (MMD format) for data from ODA database.
+
+License:
+     This file is part of the py-mmd-tools repository (https://github.com/metno/py-mmd-tools).
+     py-mmd-tools is licensed under GPL-3.0 (
+     https://github.com/metno/py-mmd-tools/blob/master/LICENSE)
+"""
+
+import json
+import yaml
+import copy
+import jinja2
+import requests
+import pathlib
+import errno
+import os
+from py_mmd_tools.xml_utils import xsd_check
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def retrieve_from_url(point, user_id='', user_pwd='', timeout_secs=10):
+    """
+    Request from an URL
+    Args:
+        point (str): URL to request to
+        user_id (str): optional, user name, default=''
+        user_pwd (str): optional, user password, default=''
+        timeout_secs (int): optional, timeout in seconds, default=10
+    Returns:
+        request object or None if the request fails.
+    """
+
+    try:
+        r = requests.get(point, auth=(user_id, user_pwd), timeout=timeout_secs)
+        logger.debug('Data retrieved OK.')
+    except requests.exceptions.RequestException as e:
+        logger.error(e)
+        return None
+    return r
+
+
+def retrieve_frost_stations(url, user_id):
+    stations_req = retrieve_from_url(url, user_id)
+    try:
+        stations_list = stations_req.json()['data']
+    except (AttributeError, KeyError, TypeError):
+        logger.error('Unable to get stations list from request.')
```

Comment:
It can fail if there's a mistake in the URL, or the ID is invalid. So the issue is actually not with the `.json()['data']` but with the request made just before. I fixed it and now handle the exception before with a `stations_req.raise_for_status()`